sbt-cross/src/main/scala/sbtcross/CrossProjectExtra.scala

```diff
@@ -0,0 +1,67 @@
+package sbtcross
+
+import java.io.File
+import scala.reflect.macros.Context
+import scala.language.experimental.macros
+
+object CrossProjectExtra {
+  private[sbtcross] def crossProject_impl(c: Context)(
+      platformsArgs: List[c.Expr[Platform]]): c.Expr[CrossProject.Builder] = {
+    import c.universe._
+
+    val enclosingValName = MacroUtils.definingValName(
+      c,
+      methodName =>
+        s"""$methodName must be directly assigned to a val, such as `val x = $methodName`.""")
+
+    val name = Literal(Constant(enclosingValName))
+
+    def javaIoFile =
+      reify { new _root_.java.io.File(c.Expr[String](name).splice) }.tree
+
+    val platforms =
+      if (!platformsArgs.isEmpty) platformsArgs.map(_.tree).toList
+      else {
+        // compatibility
+        val jsPlatform =
+          Select(Ident(newTermName("scalajscross")), newTermName("JSPlatform"))
+        val jvmPlatform =
+          Select(Ident(newTermName("sbtcross")), newTermName("JVMPlatform"))
+        List(jsPlatform, jvmPlatform)
+      }
```

Comment:
Would be nice to trigger a warning whenever this happens. You can do it from a macro via http://docs.scala-lang.org/overviews/macros/overview.html